client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -1921,6 +1925,34 @@ public CompletableFuture<KeyValueTableSegments> getCurrentSegmentsForKeyValueTab
                 LoggerHelpers.traceLeave(log, "getCurrentSegmentsForKeyValueTable", traceId);
             });
         }
+
+    @Override
+    public CompletableFuture<Map<String, List<Integer>>> getControllerToBucketMapping(BucketType bucketType) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        Preconditions.checkNotNull(bucketType, "bucketType");
+        final long requestId = requestIdGenerator.get();
+        long traceId = LoggerHelpers.traceEnter(log, "getControllerToBucketMapping", bucketType, requestId);
+
+        final CompletableFuture<ControllerToBucketMappingResponse> result = this.retryConfig.runAsync(() -> {
+            RPCAsyncCallback<ControllerToBucketMappingResponse> callback = new RPCAsyncCallback<>(requestId,
+                    "getControllerToBucketMapping", bucketType);
+
+            new ControllerClientTagger(client, timeoutMillis).withTag(requestId, GET_CONTROLLER_TO_BUCKET_MAPPING,
+                    bucketType.toString()).getControllerToBucketMapping(ControllerToBucketMappingRequest.newBuilder()
+                    .setServiceType(ControllerToBucketMappingRequest.BucketType.valueOf(bucketType.toString())).build(),
+                    callback);
+            return callback.getFuture();
+        }, this.executor);
+        return result.thenApplyAsync(response -> response.getMappingMap().entrySet().stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().getIdList())), this.executor)
+                .whenComplete((x, e) -> {
+                    if (e != null) {
+                        log.warn(requestId, "getControllerToBucketMapping request for bucket type {} failed: ",
```

Comment:
Perhaps we should let this error propagate rather than suppressing it here

Reply:
Here we are not suppressing the exception, it will propagate to caller. In case of exception, we are logging an extra warning message here.